kv/dist_sender.go

```diff
@@ -655,14 +655,17 @@ func (ds *DistSender) Send(_ context.Context, call proto.Call) {
 
 		var desc, descNext *proto.RangeDescriptor
 		err := retry.WithBackoff(retryOpts, func() (retry.Status, error) {
-			var descErr error
+			var err error
 			// Get range descriptor (or, when spanning range, descriptors).
 			// sendAttempt below may clear them on certain errors, so we
 			// refresh (likely from the cache) on every retry.
-			desc, descNext, descErr = ds.getDescriptors(call)
+			desc, descNext, err = ds.getDescriptors(call)
 			// If getDescriptors fails, we don't fish for retryable errors.
-			if descErr != nil {
-				return retry.Break, descErr
+			if err != nil {
+				if rErr, ok := err.(util.Retryable); ok && rErr != nil && rErr.CanRetry() {
```

Comment:
We're getting pretty deep in the weeds here but I think the nil check is unnecessary and arguably incorrect. It's unnecessary because returning a nil pointer wrapped in a non-nil interface value is almost certainly a bug rather than an intended effect (http://golang.org/doc/faq#nil_error), so it should be safe to assume a non-nil value here. Checking for nil may be incorrect because in the event that someone *did* do this on purpose, their `CanRetry` implementation would presumably be able to tolerate a nil receiver.